tests/test_openclhpp.cpp

```diff
@@ -4336,4 +4337,83 @@ void testTemplateGetImageRequirementsInfo()
 void testTemplateGetImageRequirementsInfo() {}
 #endif // cl_ext_image_requirements_info
 
+static cl_mem clCreateFromGLBuffer_testgetObjectInfo(cl_context context,
+                                                     cl_mem_flags flags,
+                                                     cl_GLuint bufobj,
+                                                     cl_int *errcode_ret,
+                                                     int num_calls)
+{
+    switch (num_calls) {
+    case 0:
+            TEST_ASSERT_EQUAL(0, bufobj);
+            break;
+    case 1:
+            TEST_ASSERT_EQUAL(1, bufobj);
+            break;
+    default:
+            break;
+    }
+
+    TEST_ASSERT_EQUAL_PTR(make_context(0), context);
+    TEST_ASSERT_EQUAL(0, flags);
+    if (errcode_ret)
+        *errcode_ret = CL_SUCCESS;
+    return make_mem(0);
+}
+
+static cl_int clGetGLObjectInfo_testgetObjectInfo(cl_mem memobj,
+                                                  cl_gl_object_type *type,
+                                                  cl_GLuint *gl_object_name,
+                                                  int num)
+{
+    cl_int ret = CL_SUCCESS;
+    TEST_ASSERT_EQUAL(memobj, make_mem(0));
+    switch (num) {
+    case 0:
+        TEST_ASSERT_EQUAL(CL_GL_OBJECT_TEXTURE2D_ARRAY, *type);
+        *type = CL_GL_OBJECT_BUFFER;
+        break;
+    case 1:
+        TEST_ASSERT_EQUAL(CL_GL_OBJECT_BUFFER, *type);
+        *type = CL_GL_OBJECT_TEXTURE2D_ARRAY;
+        ret = CL_DEVICE_NOT_FOUND;
```

Comment:
There are a couple of problems here:

1. The `type` is an output value, so we really don't care what gets passed in for it.  This check should be on the caller, not in this stub.
2. We shouldn't be returning an error from this function because it causes the executables with exceptions enabled to throw an exception, which causes the test to fail.  Unless an error condition really is occurring for some reason - and it isn't in any of these tests - we shoudl return `CL_SUCCESS`.

Reply:
Fixed